### PR TITLE
 Implement OGC Coverages scaling requirements class

### LINKED ISSUE #914: Implement OGC Coverages scaling requirements class
The requirements class is defined [here](https://docs.ogc.org/DRAFTS/19-087.html#rc-scaling). Currently we have a partial implementation, supporting only the parameter `scale-factor`.

### PR DESCRIPTION
**Pull request by pont-us**
Closes #914 .

Fully implement OGC Coverages scaling requirements class.

In practice, this means that we now support the `scale-size` and `scale-axes` parameters as well as the previously supported `scale factor`.

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: xcube/webapi/ows/coverages/scaling.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                map(
+                    lambda attr: getattr(axis, attr, '').lower(),
+                    ['name', 'abbrev'],
+                )
+            )
+            if identifiers & valid_identifiers:
+                return axis.abbrev
+        return None
+
+    def apply(self, gm: GridMapping):
```
Comment (TonioF) [inline xcube/webapi/ows/coverages/scaling.py:127]:
```suggestion
    def apply(self, gm: GridMapping) -> GridMapping:
```

Also, a tiny bit of documentation would be nice

File: test/webapi/ows/coverages/test_scaling.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            CoverageScaling(
+                CoverageRequest({}), CrsMock(), self.ds
+            ).get_axis_from_crs(set())
+        )
+
+    def test_scale_factor(self):
+        scaling = CoverageScaling(
+            CoverageRequest({'scale-factor': ['2']}), self.epsg4326, self.ds
+        )
+        self.assertEqual((2, 2), scaling.scale)
```
Comment (TonioF) [inline test/webapi/ows/coverages/test_scaling.py:45]:
```suggestion
        self.assertEqual((2, 2), scaling.scale)
        self.assertEqual((180, 90), scaling.size)
```

File: test/webapi/ows/coverages/test_scaling.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    def test_scale_size(self):
+        scaling = CoverageScaling(
+            CoverageRequest({'scale-size': ['Lat(90),Lon(240)']}),
+            self.epsg4326,
+            self.ds,
+        )
+        self.assertEqual((240, 90), scaling.size)
+        self.assertEqual((1.5, 2), scaling.scale)
+
+    def test_apply_identity_scaling(self):
```
Comment (TonioF) [inline test/webapi/ows/coverages/test_scaling.py:65]: A test that tests the non-identity case would have been nice.

**Review by TonioF — CHANGES_REQUESTED** [review]
A few minor comments.

File: test/webapi/ows/coverages/test_scaling.py (hunk 3 above)
Comment (pont-us) [reply test/webapi/ows/coverages/test_scaling.py:65]: The tests above this one (`test_scale_factor`, `test_scale_axes`, and `test_scale_size`) already test various non-identity cases using all the available scaling parameters.

File: xcube/webapi/ows/coverages/controllers.py [hunk 4]
```diff
@@ -440,7 +424,9 @@ def _correct_inverted_y_range_if_necessary(
 
 
 def dataset_to_image(
-    ds: xr.Dataset, image_format: Literal['png', 'tiff'] = 'png'
+    ds: xr.Dataset,
+    image_format: Literal['png', 'tiff'] = 'png',
+    crs: pyproj.CRS = None,
```
[the remark this answers is not in the document]
Comment (pont-us) [reply xcube/webapi/ows/coverages/controllers.py:429]: Done.

File: xcube/webapi/ows/coverages/util.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+def get_h_dim(ds: xr.Dataset):
+    return [
+        d for d in list(map(str, ds.dims)) if d[:3].lower() in {'x', 'lon'}
+    ][0]
+
+
+def get_v_dim(ds: xr.Dataset):
+    return [
+        d for d in list(map(str, ds.dims)) if d[:3].lower() in {'y', 'lat'}
+    ][0]
```
[the remark this answers is not in the document]
Comment (pont-us) [reply xcube/webapi/ows/coverages/util.py]: Done.

File: xcube/webapi/ows/coverages/scaling.py (hunk 1 above)
Comment (pont-us) [reply xcube/webapi/ows/coverages/scaling.py]: Type hint added, along with docstrings and comments for the whole class -- the implementation is simple but has one or two subtleties that are worth spelling out, so thanks for the reminder to document.

File: test/webapi/ows/coverages/test_scaling.py (hunk 2 above)
Comment (pont-us) [reply test/webapi/ows/coverages/test_scaling.py]: Done.

**Pull request merged**